.github/workflows/accSelector.yaml

```diff
@@ -0,0 +1,46 @@
+name: SetM365AppAcc
+
+on:
+  workflow_call:
+    outputs:
+      client_app_slot:
+        value: ${{ jobs.GetM365App.outputs.client_app_slot }}
+      client_id_env:
+        value: ${{ jobs.GetM365App.outputs.client_id_env }}
+      client_secret_env:
+        value: ${{ jobs.GetM365App.outputs.client_secret_env }}
+
+jobs:
+  GetM365App:
+    environment: Testing
+    runs-on: ubuntu-latest
+    outputs:
+      client_app_slot: ${{ steps.roundrobin.outputs.CLIENT_APP_SLOT }}
+      client_id_env: ${{ steps.roundrobin.outputs.CLIENT_ID_ENV }}
+      client_secret_env: ${{ steps.roundrobin.outputs.CLIENT_SECRET_ENV }}
+    steps:
+      - name: Figure out which client id to use
+        id: roundrobin
+        run: |
+          slot=$((GITHUB_RUN_NUMBER % 4))
+          echo "CLIENT_APP_SLOT=$slot" >> $GITHUB_OUTPUT
+
+          case $slot in
+
+            0)
+              echo "CLIENT_ID_ENV=CLIENT_ID" >> $GITHUB_OUTPUT
```

Comment:
nit: The numbering is off by 1. Maybe change CLIENT_ID_2 to CLIENT_ID_1 and so on?

Reply:
These are the existing Github secrets. Didn't want to change that. So went with existing names